components/NavigationBar/tests/unit/NavigationBarTitleViewTests.swift

```diff
@@ -0,0 +1,140 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+import MaterialComponents.MaterialNavigationBar
+
+class NavigationBarTitleViewTests: XCTestCase {
+
+  // MARK: Direct assignment
+
+  func testTitleViewAssignmentAddsItAsASubview() {
+    // Given
+    let navigationBar = MDCNavigationBar()
+    let titleView = UIView()
+
+    // When
+    navigationBar.titleView = titleView
+
+    // Then
+    XCTAssertEqual(titleView.superview, navigationBar)
+  }
+
+  func testTitleViewAssignmentTwiceStillAddsItAsASubview() {
+    // Given
+    let navigationBar = MDCNavigationBar()
+    let titleView = UIView()
+
+    // When
+    navigationBar.titleView = titleView
+    navigationBar.titleView = titleView
+
+    // Then
+    XCTAssertEqual(titleView.superview, navigationBar)
+  }
+
+  func testTitleViewAssignmentAndThenRemovalFromViewHierarchyLeavesItWithoutASuperview() {
+    // Given
+    let navigationBar = MDCNavigationBar()
+    let titleView = UIView()
+
+    // When
+    navigationBar.titleView = titleView
+    titleView.removeFromSuperview()
+
+    // Then
+    XCTAssertNil(titleView.superview)
+  }
+
+  func testTitleViewAssignmentThenRemovalFromViewHierarchyThenReassignmentAddsItAsASubview() {
+    // Given
+    let navigationBar = MDCNavigationBar()
+    let titleView = UIView()
+
+    // When
+    navigationBar.titleView = titleView
+    titleView.removeFromSuperview()
+    navigationBar.titleView = titleView
+
+    // Then
+    XCTAssertEqual(titleView.superview, navigationBar)
+  }
+
+  // MARK: Assignment view UINavigationItem
+
+  func testNavigationItemTitleViewAssignmentBeforeObservationAddsTitleViewAsASubview() {
+    // Given
+    let navigationBar = MDCNavigationBar()
+    let titleView = UIView()
+    let navigationItem = UINavigationItem()
+
+    // When
+    navigationItem.titleView = titleView
+    navigationBar.observe(navigationItem)
+
+    // Then
+    XCTAssertEqual(titleView.superview, navigationBar)
+  }
+
+  func testNavigationItemTitleViewAssignmentAfterObservationAddsTitleViewAsASubview() {
+    // Given
+    let navigationBar = MDCNavigationBar()
+    let titleView = UIView()
+    let navigationItem = UINavigationItem()
+
+    // When
+    navigationBar.observe(navigationItem)
+    navigationItem.titleView = titleView
+
+    // Then
+    XCTAssertEqual(titleView.superview, navigationBar)
+  }
+
+  func testNavigationItemTitleViewAssignmentWithSimulatedTheftKeepsTitleViewAsSubview() {
+    // Given
+    let navigationBar = MDCNavigationBar()
+    let titleView = UIView()
+    let navigationItem = UINavigationItem()
+
+    // When
+    navigationItem.titleView = titleView
+    navigationBar.observe(navigationItem)
+    let simulatedThiefView = UIView()
```

Comment:
should this instantiation be moved to Given?

Reply:
Done.